azurelinuxagent/ga/exthandlers.py

```diff
@@ -263,6 +263,14 @@ def run(self):
                       message=detailed_msg)
             return
 
+    def changed(self):
```

Comment:
You mentioned earlier that there is some scenario where you need to check if goal_state incarnation changed or not which is independent of `ext_config.changed`. Not sure if that check is needed here too or not.

Reply:
self.ext_config.changed factors in the incarnation